### PR TITLE
Fixes for typos / improvements for English

### PR DESCRIPTION
**Pull request by declension**
* Fix some typos
* Fix / Improve English for readability in a few places 
* Use Blockquote for quotes
* Also add semantic links

### CODE REVIEW

File: expression-problem.md [hunk 1]
```diff
@@ -191,24 +190,24 @@ main = do
...
 
 ## Conclusion
-"Tagless Final" approach seems like a good way to completely solve Expression Problem in Haskell, although I am not yet convinced practical is the solution in real world use cases (but I haven't understood / tested it out well enough so that is why).
+The "Tagless Final" approach seems like a good way to completely solve Expression Problem in Haskell, although I am not yet convinced practical is the solution in real world use cases (but I haven't understood / tested it out well enough so that is why).
 
-On the other hand, simpler approach I used above almost solves it (requires updating of just one line when new operation is added), I find it more intuitive, and I am more assured of its practicality in real world use cases, so I would probably pick that one for now if need be.
+On the other hand, the simpler approach I used above almost solves it (requires updating of just one line when a new operation is added). I find it more intuitive, and I am more assured of its practicality in real world use cases, so I would probably pick that one for now if need be.
```
Comment (Martinsos) [inline expression-problem.md]:
```suggestion
On the other hand, the simpler approach I used above almost solves it (requires updating of just one line when a new operation is added), I find it more intuitive, and I am more assured of its practicality in real world use cases, so I would probably pick that one for now if need be.
```

File: expression-problem.md [hunk 2]
```diff
@@ -1,25 +1,25 @@
-Wikipedia definition: https://en.wikipedia.org/wiki/Expression_problem  
+[Wikipedia definition of the Expression Problem](https://en.wikipedia.org/wiki/Expression_problem) 
 TLDR; How do you make your code open to both adding new operations and new types without modifying existing code?
 
-In OOP, by default it is easy to add new types (add new class that implements existing operations) but to add new operation you need to modify the existing classes.  
-In FP, by default it is easy to add new operation (new new function that pattern matches on existing types) but to add new type you need to modify the existing operations.
+In OOP, by default it is easy to add new types (add new class that implements existing operations) but to add new operations you need to modify the existing classes.  
+In FP, by default it is easy to add new operation (new new function that pattern matches on existing types) but to add new types you need to modify the existing operations.
```
Comment (Martinsos) [inline expression-problem.md]:
```suggestion
In FP, by default it is easy to add new operation (new function that pattern matches on existing types) but to add new types you need to modify the existing operations.
```

**Review by Martinsos — APPROVED** [review]
Great, thanks for fixes :)!
